clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -888,11 +889,18 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 // treat subscribing to empty topic list as the same as unsubscribing
                 this.unsubscribe();
             } else {
+                IllegalArgumentException iae = new IllegalArgumentException("Detected issues with topic names in topic partitions to assign:");
```

Comment:
Can we use throwIfTopicNotValid private method to avoid duplicate code?

Reply:
If we throw InvalidTopicException directly, be change public API (possible exception are part of the API) and thus, this would require a KIP.

Reply:
I think @omkreddy is just suggesting to create a helper to avoid the code duplication. Seems like a good idea to me.